lib/shopify-cli/commands/connect.rb

```diff
@@ -9,39 +9,23 @@ def call(args, command_name)
           super
         end
 
-        def default_connect(project_type)
-          org = ShopifyCli::Tasks::EnsureEnv.call(@ctx, regenerate: true)
-          write_cli_yml(project_type, org['id']) unless Project.has_current?
-          api_key = Project.current(force_reload: true).env['api_key']
-          [org, api_key]
-        end
-
-        def write_cli_yml(project_type, org_id)
-          ShopifyCli::Project.write(
-            @ctx,
-            project_type: project_type,
-            organization_id: org_id,
-          )
-          @ctx.done(@ctx.message('core.connect.cli_yml_saved'))
-        end
-
         def help
           ShopifyCli::Context.message('core.connect.help', ShopifyCli::TOOL_NAME)
         end
       end
 
       def call(args, command_name)
-        if Project.has_current? && Project.current && Project.current.env
+        if Project.current&.env
           @ctx.puts(@ctx.message('core.connect.already_connected_warning'))
         end
 
         project_type = ask_project_type
 
-        klass = ProjectType.load_type(project_type).connect_command
-        klass.ctx = @ctx
-        if klass
+        begin
+          klass = ProjectType.load_type(project_type).connect_command
```

Comment:
I'm not a fan of using the exception to direct our flow here, as I worry it may hide actual errors. Is there a check we can run to go to default instead to those errors visible?